src/main/kotlin/ink/abb/pogo/scraper/Bot.kt

```diff
@@ -261,4 +267,28 @@ class Bot(val api: PokemonGo, val settings: Settings) {
     fun task(task: Task) {
         task.run(this, ctx, settings)
     }
+
+    fun checkForPlannedStop():Boolean {
+        val timeDiff:Long = ChronoUnit.MINUTES.between(ctx.startTime, LocalDateTime.now())
+        val pokemonCatched:Int = ctx.pokemonStats.first.get()
+        val pokestopsVisited:Int = ctx.pokestops.get()
+        //Log.red("time: ${timeDiff}, pokemon: ${pokemonCatched}, pokestops: ${pokestopsVisited}")
+        if(settings.botTimeoutAfterMinutes <= timeDiff && settings.botTimeoutAfterMinutes != -1){
+            Log.red("Bot timed out as declared in the settings (after ${settings.botTimeoutAfterMinutes} minutes)")
+            return true
+        } else if(settings.botTimeoutAfterCatchingPokemon <= pokemonCatched && settings.botTimeoutAfterCatchingPokemon != -1){
+            Log.red("Bot timed out as declared in the settings (after catching ${settings.botTimeoutAfterCatchingPokemon} pokemon)")
+            return true
+        } else if(settings.botTimeoutAfterVisitingPokestops <= pokestopsVisited && settings.botTimeoutAfterVisitingPokestops != -1){
+            Log.red("Bot timed out as declared in the settings (after visiting ${settings.botTimeoutAfterVisitingPokestops} pokestops)")
+            return true
+        }
+        return false
+    }
+
+    fun terminateApplication(){
+        phaser.forceTermination()
+        ProgramController.stopAllApplications()
```

Comment:
but we don't run it as different process, we have multiple, json files for the various instances so you must not terminate these all but only the one reached the limit